mitodl/ocw-studio
# Google Drive integration

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #408

#### What's this PR do?
Adds tasks, models and functions to check for new video files in a particular google team drive, and if a matching website is found based on the parent folder(s), upload them to S3.

#### How should this be manually tested?
- Set the following in your `.env` file:
  ```
  DRIVE_SERVICE_ACCOUNT_CREDS=<ask me>
  DRIVE_SHARED_ID=<ask me>
  AWS_STORAGE_BUCKET_NAME* = <same as on RC, ditto for AWS keys>
  ```
- Create or choose an existing `Website`, note the `short_id`, and create a folder in the team drive with the `short_id` as the name.  Create a subfolder under that.  Then upload a video file to that subfolder.
- Wait.  Within a few minutes, a celery task should run, create a `DriveFile` object for that video, and upload the video to S3.


#### Where should the reviewer start?
#403 

#### Any background context you want to provide?
The original plan was to create a Google Drive push notification that would trigger whenever a file was added/modified, and add a webhook endpoint in studio to process that.  But there seems to be a bug with team drives that prevents the notification from being triggered unless the file is added/modified by the service account that created the push notification.
